jobboard/views.py

```diff
@@ -38,13 +37,38 @@ def get_content_if_user_is_not_student(request, user_is_recruiter):
     return content
 
 
+def searched_by_companies_or_keywords(searched_name):
+    jobs_by_company = Job.get_jobs_by_company_name(searched_name)
+    jobs_by_keywords = Job.get_jobs_by_keywords(searched_name)
+    return jobs_by_company.union(jobs_by_keywords)
+
+
+def searched_by_area(searched_area):
+    jobs_by_city = Job.get_jobs_by_city_name(searched_area)
+    return jobs_by_city
+
+
 def board(request):
     user_is_student = Student.is_student(request.user.id)
     user_is_recruiter = Recruiter.is_recruiter(request.user.id)
     if user_is_student:
         content = get_content_if_user_is_student(request)
     else:
         content = get_content_if_user_is_not_student(request, user_is_recruiter)
+
+    # user hit the search button
+    if request.method == "POST":
+        searched_name = request.POST.get('searched_name')
+        searched_area = request.POST.get('searched_area')
+        content["searched_name"] = searched_name
+        content["searched_area"] = searched_area
+        if searched_name and not searched_area:
+            content['jobs_to_display'] = searched_by_companies_or_keywords(searched_name)
+        elif searched_area and not searched_name:
+            content['jobs_to_display'] = searched_by_area(searched_area)
+        else:
+            content['jobs_to_display'] = searched_by_area(searched_area).\
+                intersection(searched_by_companies_or_keywords(searched_name))
     add_navbar_links_to_context(request, content)
     return render(request, 'jobboard/board.html', content)
```

Comment:
I think we should consider the case in which non-job corresponds to the search. Can maybe be implemented by  initializing jobs_to_display to None, then doing the same logic you have implemented with the "if.. eleif..", and then to add something like 
`if  content['jobs_to_display'] is None:
      return render(request, 'non_job_matched_the_search.html') `
 and also add that "non_job_matched_the_search.html" page

Reply:
I don't know if another html page is necessary but I can display a text saying "No jobs matched this search" or something like that.